internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -124,3 +144,147 @@ func (m *DynamicScaleUpManager) reconcileClustersForRegions() []error {
+//        For the calculation of the max capacity:
+//        * Clusters in deprovisioning and cleanup state are excluded, as
+//          clusters into those states don't accept kafka instances anymore.
+//        * Clusters that are still not ready to accept kafka instance but that
+//          should eventually accept them (like accepted state for example)
+//          are included
+// Otherwise false is returned.
+// Note: Clusters in the 'failed' state are excluded from the calculations.
+func (m *DynamicScaleUpManager) shouldScaleUpNewClusterForInstanceType(
+	provider string, region string,
+	instanceTypeName string,
+	instanceTypeConfig *config.InstanceTypeConfig,
+	kafkaStreamingUnitCountPerClusterList services.KafkaStreamingUnitCountPerClusterList) (bool, error) {
+
+	kafkaInstanceTypeConfig, err := m.KafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(instanceTypeName)
+	if err != nil {
+		return false, err
+	}
+	biggestKafkaInstanceSizeCapacityConsumption := -1
+	maxKafkaInstanceSizeConfig := kafkaInstanceTypeConfig.GetBiggestCapacityConsumedSize()
+	if maxKafkaInstanceSizeConfig != nil {
+		biggestKafkaInstanceSizeCapacityConsumption = maxKafkaInstanceSizeConfig.CapacityConsumed
+	}
+
+	consumedStreamingUnitsInRegion := 0
+	maxStreamingUnitsInRegion := 0
+	atLeastOneClusterHasCapacityForLargestInstanceType := false
+
+	scaleUpActionIsOngoing := false
+	clusterStatesTowardReadyState := []string{
+		api.ClusterProvisioning.String(), api.ClusterProvisioned.String(),
+		api.ClusterAccepted.String(), api.ClusterWaitingForKasFleetShardOperator.String(),
+	}
+
+	for _, kafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
+		if kafkaStreamingUnitCountPerCluster.CloudProvider == provider &&
+			kafkaStreamingUnitCountPerCluster.Region == region &&
+			kafkaStreamingUnitCountPerCluster.InstanceType == instanceTypeName {
+
+			if arrays.Contains(clusterStatesTowardReadyState, kafkaStreamingUnitCountPerCluster.Status) {
+				scaleUpActionIsOngoing = true
+			}
+
+			if kafkaStreamingUnitCountPerCluster.FreeStreamingUnits() >= int32(biggestKafkaInstanceSizeCapacityConsumption) {
+				atLeastOneClusterHasCapacityForLargestInstanceType = true
+			}
+
+			consumedStreamingUnitsInRegion = consumedStreamingUnitsInRegion + int(kafkaStreamingUnitCountPerCluster.Count)
+			if kafkaStreamingUnitCountPerCluster.Status != api.ClusterDeprovisioning.String() &&
+				kafkaStreamingUnitCountPerCluster.Status != api.ClusterCleanup.String() {
+				maxStreamingUnitsInRegion = maxStreamingUnitsInRegion + int(kafkaStreamingUnitCountPerCluster.MaxUnits)
+			}
+		}
+	}
+
+	freeStreamingUnitsInRegion := maxStreamingUnitsInRegion - consumedStreamingUnitsInRegion
+	capacitySlackInRegion := instanceTypeConfig.MinAvailableCapacitySlackStreamingUnits
+	notEnoughCapacitySlackInRegion := false
+	if freeStreamingUnitsInRegion < capacitySlackInRegion { // TODO is it <= or < ?
```

Comment:
> // TODO is it <= or < ?

I think `<` because we want to be inclusive. i.e if free capacity is `4` and slack is `4`, there is no need to trigger a scale up in my opinion.

Reply:
If it is `<` then if the capacity slack is not set then it would never perform data plane scale up, asfreeStreamingUnitsInRegion would never be less than 0 . Do we want to define it as that?

Reply:
That will depend on the other conditions right?

Let me try to enumerate the cases:
- region limits have not been reached and no scale up action is ongoing.
- capacity slack is `0` and freeCapacityInRegion is greater than `0`. Under this `notEnoughCapacitySlackInRegion` will be `false`
- let's assume that `atLeastOneClusterHasCapacityForLargestInstanceType` is `false` (because in the total free capacity in the region is not enough to accomodate the largest kafka), under this condition we need to perform scale up.

Reply:
It shouldn't depend on any other conditions. We should deal with that condition independently.

The question/condition is: "is there enough capacity slack or not?" independently on the other conditions

Reply:
As discussed directly, having `<` makes sense. The behavior would be:
* If the configured capacity slack is 0 or unset that means that no matter the value of free capacity the outcome will be that there is always capacity slack (notEnoughCapacitySlackInRegion will always be false). This means that in that case the only scenario where data plane scale up will be performed is the scenario where the largest kafka size don't fit in any single cluster in the region
* If the configured capacity slack is > 0 then notEnoughCapacitySlackInRegion will be set to true when the free capacity is less than the configured slack

Reply:
Thanks @miguelsorianod for the summary. +1